src/main/java/nextstep/subway/domain/Section.java

```diff
@@ -62,4 +62,21 @@ public void update(Station newUpStation, int minusDistance) {
         this.distance = this.distance - minusDistance;
     }
 
+    public Section merge(Section section) {
+        if (!isDownStation(section.upStation)) {
+            throw new IllegalArgumentException("합치려는 구간의 상행역이 하행역과 같아야 합니다.");
+        }
+
+        return Section.of(line, upStation, section.downStation, distance + section.distance);
+    }
+
+
+    public boolean isUpStation(Station station) {
```

Comment:
지금 네이밍도 좋지만 의미가 조금 더 명확히 들어날 수 있게 아래처럼 네이밍 해 볼 수도 있겠네요 :) 
```suggestion
    public boolean hasUpStationAs(Station station) {
```

이렇게 네이밍을 변경했을 때 메서드를 호출하는 쪽에서는 아래와 같이 호출할 수 있습니다.
```
section.hasUpStationAs(station)
```